wp-content/civi-extensions/goonjcustom/Civi/MaterialContributionService.php

```diff
@@ -67,6 +68,10 @@ public static function attachContributionReceiptToEmail(&$params, $context = NUL
 
     $contribution = $activities->first();
 
+    $goonjOfficeId = $contribution['Material_Contribution.Goonj_Office'] ?? null;
+
+    $city = self::getCityFromGoonjOfficeId($goonjOfficeId);
```

Comment:
You should simply the job of this function by doing this:
```suggestion
    $city = self::getContributionCity($contribution);
```

Then you should create a function

```php
private static function getContributionCity($contribution) {
  $officeId = $contribution['Material_Contribution.Goonj_Office'];


  if (!$officeId) {
    // check for collection camp
    // if no collection camp 
     return '';
   else {
        return city of collection camp
   }
  }

  // if we are here then it means there is an office id
  // fetch the city of the office and return it
}
```

Reply:
done changes